### PR TITLE
chore: Add information on Security Considerations to constructs

### PR DESCRIPTION
**Pull request by ddneilson**
This is adding information on security considerations that customers should take into account when using the RFDK constructs. These are concrete recommendations, rather than security posture-type information.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by grbartel — APPROVED** [review]

**Review by jusiskin — CHANGES_REQUESTED** [review]
This will be great for RFDK users. Just some minor additional considerations I think would be beneficial to add.

File: packages/aws-rfdk/lib/core/lib/health-monitor.ts [hunk 1]
```diff
@@ -257,12 +257,30 @@ abstract class HealthMonitorBase extends Construct implements IHealthMonitor {
...
+ * - The AWS Lambda that is deployed through this construct will be created from a deployment package
+ *   that is uploaded to your CDK bootstrap bucket during deployment. You must limit write access to
+ *   your CDK bootstrap bucket to prevent an attacker from modifying the actions performed by this Lambda.
+ *   We strongly recommend that you either enable Amazon S3 server access logging on your CDK bootstrap bucket,
+ *   or enable AWS CloudTrail on your account to assist in post-incident analysis of compromised production
+ *   environments.
+ * - The AWS Lambda that is created by this construct to terminate unhealthy worker fleets has permission to
+ *   UpdateAutoScalingGroup ( https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_UpdateAutoScalingGroup.html )
+ *   on all of the fleets that this construct is monitoring. You should not grant any additional actors/principals the
+ *   ability to modify or execute this Lambda.
```
[the remark this answers is not in the document]
Comment (ddneilson) [reply packages/aws-rfdk/lib/core/lib/health-monitor.ts:279]: That seems a little out-there to me. Probably not necessary

File: packages/aws-rfdk/lib/core/lib/mongodb-instance.ts [hunk 2]
```diff
@@ -304,11 +304,24 @@ export interface IMongoDb extends IConnectable, IConstruct {
...
- * - An A Record in the provided PrivateHostedZone to create a DNS entry for this server's static private IP.
- * - A Secret containing the administrator credentials for MongoDB.
- * - An encrypted EBS Volume on which the MongoDB data is stored.
+ * - An A-Record in the provided PrivateHostedZone to create a DNS entry for this server's static private IP.
+ * - A Secret in AWS SecretsManager that contains the administrator credentials for MongoDB.
+ * - An encrypted Amazon Elastic Block Store (EBS) Volume on which the MongoDB data is stored.
  * - Amazon CloudWatch log group that contains instance-launch and MongoDB application logs.
  *
+ * Security Considerations
+ * ------------------------
```
[the remark this answers is not in the document]
Comment (ddneilson) [reply packages/aws-rfdk/lib/core/lib/mongodb-instance.ts:313]: Re: Backup -- it's in the docstring for this construct: "it is recommended that you set up a backup schedule for this volume."

I agree about adding a call-out about securing the EBS volume. Added for the next revision.

File: packages/aws-rfdk/lib/core/lib/staticip-server.ts [hunk 3]
```diff
@@ -159,12 +159,24 @@ export interface StaticPrivateIpServerProps {
...
- * 6) Lambda function, with role, to attach the ENI in response to instance-launch lifecycle events -- max one per stack.
+ * - Auto Scaling Group (ASG) with min & max capacity of 1 instance;
+ * - Elastic Network Interface (ENI);
+ * - Security Group for the ASG;
+ * - Instance Role and corresponding IAM Policy
+ * - SNS Topic & Role for instance-launch lifecycle events -- max one of each per stack; and
+ * - Lambda function, with role, to attach the ENI in response to instance-launch lifecycle events -- max one per stack.
+ *
+ * Security Considerations
+ * ------------------------
```
[the remark this answers is not in the document]
Comment (ddneilson) [reply packages/aws-rfdk/lib/core/lib/staticip-server.ts:170]: Yeah, that would be a good idea.

File: packages/aws-rfdk/lib/core/lib/x509-certificate.ts [hunk 4]
```diff
@@ -236,9 +236,18 @@ abstract class X509CertificateBase extends Construct {
...
  * ------------------------
- * 1) DynamoDB Table - Used for tracking resources created by the Custom Resource.
- * 2) Secrets - 4 in total, for the certificate, it's private key, the passphrase to the key, and the cert chain.
- * 3) Lambda Function, with role - Used to create/update/delete the Custom Resource
+ * - DynamoDB Table - Used for tracking resources created by the Custom Resource.
+ * - Secrets - 4 in total, for the certificate, it's private key, the passphrase to the key, and the cert chain.
+ * - Lambda Function, with role - Used to create/update/delete the Custom Resource
+ *
+ * Security Considerations
+ * ------------------------
```
[the remark this answers is not in the document]
Comment (ddneilson) [reply packages/aws-rfdk/lib/core/lib/x509-certificate.ts:244]: Absolutely. I can't believe that I missed that!

File: packages/aws-rfdk/lib/deadline/lib/repository.ts [hunk 5]
```diff
@@ -347,18 +347,25 @@ export interface RepositoryProps {
...
- * 5) A Script Asset which is uploaded to your deployment bucket to run the installer
- * 6) An aws-rfdk.CloudWatchAgent to configure sending logs to cloudwatch.
+ * - Encrypted Amazon Elastic File System (EFS) - If no file system is provided;
+ * - An Amazon DocumentDB - If no database connection is provided;
+ * - Auto Scaling Group (ASG) with min & max capacity of 1 instance;
+ * - Instance Role and corresponding IAM Policy
+ * - An Amazon CloudWatch log group that contains the Deadline Repository installation logs
+ *
+ * Security Considerations
+ * ------------------------
```
[the remark this answers is not in the document]
Comment (ddneilson) [reply packages/aws-rfdk/lib/deadline/lib/repository.ts:363]: Yup. Not sure how I missed mentioning that...

File: packages/aws-rfdk/lib/deadline/lib/usage-based-licensing.ts [hunk 6]
```diff
@@ -440,10 +440,21 @@ export interface UsageBasedLicensingProps {
...
+ * Security Considerations
+ * ------------------------
+ * - The instances deployed by this construct download and run scripts from your CDK bootstrap bucket when that instance
+ *   is launched. You must limit write access to your CDK bootstrap bucket to prevent an attacker from modifying the actions
+ *   performed by these scripts. We strongly recommend that you either enable Amazon S3 server access logging on your CDK
+ *   bootstrap bucket, or enable AWS CloudTrail on your account to assist in post-incident analysis of compromised production
+ *   environments.
+ * - The Deadline License Forwarder is designed to be secured by restricting network access to it. For security, only the Deadline
+ *   Workers that require access to Usage-based Licenses should be granted network access to the instances deployed by this construct.
+ *   Futhermore, you should restrict that access to only the product(s) that those workers require when deploying this construct.
```
[the remark this answers is not in the document]
Comment (ddneilson) [reply packages/aws-rfdk/lib/deadline/lib/usage-based-licensing.ts:457]: It doesn't actually impact the UBL credits -- those are a different mechanism that does not involve the software deployed by this construct. Ultiimately, the risk is piracy. I think what's here is sufficient.

File: packages/aws-rfdk/lib/deadline/lib/worker-fleet.ts [hunk 7]
```diff
@@ -279,10 +279,18 @@ abstract class WorkerInstanceFleetBase extends Construct implements IWorkerFleet
...
- * 2) An Instance Role and corresponding IAM Policy;
- * 3) A script asset which is uploaded to your deployment bucket used to configure the worker so it can connect to the Render Queue
- * 4) An aws-rfdk.CloudWatchAgent to configure sending logs to cloudwatch.
+ * - An EC2 Auto Scaling Group to maintain the number of instances;
+ * - An Instance Role and corresponding IAM Policy;
+ * - An Amazon CloudWatch log group that contains the Deadline Worker, Deadline Launcher, and instance-startup logs for the instances
+ *   in the fleet.
+ *
+ * Security Considerations
+ * ------------------------
```
[the remark this answers is not in the document]
Comment (ddneilson) [reply packages/aws-rfdk/lib/deadline/lib/worker-fleet.ts:288]: I added something on patching. I'm not sure if it's worth mentioning that the instance has unrestricted access to Deadline. That kind of seems obvious

**Comment by ddneilson** [comment]
I need to rebuild this PR. The switch to public repo severed the link between this and my repo.

**Pull request closed without merging**